src/System.Linq.Expressions/src/System/Linq/Expressions/MethodCallExpression.cs

```diff
@@ -879,6 +1042,34 @@ public static MethodCallExpression Call(Type type, string methodName, Type[] typ
         ///<paramref name="instance" />.Type is not assignable to the declaring type of the method represented by <paramref name="method" />.-or-The number of elements in <paramref name="arguments" /> does not equal the number of parameters for the method represented by <paramref name="method" />.-or-One or more of the elements of <paramref name="arguments" /> is not assignable to the corresponding parameter for the method represented by <paramref name="method" />.</exception>
         public static MethodCallExpression Call(Expression instance, MethodInfo method, IEnumerable<Expression> arguments)
         {
+            IReadOnlyList<Expression> argumentList = arguments as IReadOnlyList<Expression>;
+
+            if (argumentList != null)
+            {
+                switch (argumentList.Count)
+                {
+                    case 0:
+                        return Call(instance, method);
+                    case 1:
+                        return Call(instance, method, argumentList[0]);
+                    case 2:
+                        return Call(instance, method, argumentList[0], argumentList[1]);
+                    case 3:
+                        return Call(instance, method, argumentList[0], argumentList[1], argumentList[2]);
+                }
+
+                if (instance == null)
```

Comment:
Note there's always the option to go beyond optimized instance call nodes with 3 parameters. I've run a few LINQ queries over all public types in mscorlib, System, System.Core on Windows 10 and here are the stats:

Static methods:

{ Arity = 0, Count = 553 }
{ Arity = 1, Count = 1724 }
{ Arity = 2, Count = 1450 }
{ Arity = 3, Count = 504 }
{ Arity = 4, Count = 231 }
{ Arity = 5, Count = 118 }
{ Arity = 6, Count = 38 }
{ Arity = 7, Count = 18 }
{ Arity = 8, Count = 6 }
{ Arity = 9, Count = 7 }
{ Arity = 10, Count = 3 }
{ Arity = 11, Count = 1 }
{ Arity = 12, Count = 1 }

Instance methods:

{ Arity = 0, Count = 39542 }
{ Arity = 1, Count = 20498 }
{ Arity = 2, Count = 5331 }
{ Arity = 3, Count = 2032 }
{ Arity = 4, Count = 780 }
{ Arity = 5, Count = 625 }
{ Arity = 6, Count = 171 }
{ Arity = 7, Count = 81 }
{ Arity = 8, Count = 131 }
{ Arity = 9, Count = 27 }
{ Arity = 10, Count = 30 }
{ Arity = 11, Count = 4 }
{ Arity = 12, Count = 4 }
{ Arity = 13, Count = 5 }
{ Arity = 14, Count = 4 }
{ Arity = 15, Count = 4 }
{ Arity = 16, Count = 4 }
{ Arity = 17, Count = 2 }
{ Arity = 18, Count = 2 }

The addition for low argument counts on instance methods seems well worth it. An argument count of 4 and 5 is still quite frequent for those as well.